server/graphql/v2/query/index.js

```diff
@@ -32,6 +33,7 @@ const query = {
   transactions: TransactionsQuery,
   update: UpdateQuery,
   paypalPlan: PaypalPlanQuery,
+  latestChangelogPublishDate: LatestChangelogPublishDateQuery,
```

Comment:
I find this query too specific and would like to challenge the design here.

Better use a generic query that returns a collection of updates and:
- make sure that there is a boolean filter to request changelog entries
- make sure that they are ordered ante-chronologically by default 
- make sure that there is pagination and one can ask a limit of 1

Then you have exactly what you need with a GraphQL query:

```
 query LatestChangelogPublishDateQuery($slug: String!) {
    account(slug: $slug) {
     updates(isChangelog: true, limit: 1) {
        nodes {
           id
          createdAt
    }
```

or something like:


```
 query LatestChangelogPublishDateQuery($account: AccountReferenceInput!) {
     updates(account: $account, isChangelog: true, limit: 1) {
        nodes {
           id
          createdAt
    }
```

Reply:
@znarf : Thanks for the input. A small doubt I have with this approach is that the caching of the latest change log publish date. With this approach, we could cache the result of this query based on the limit being equal to 1 and that it's a changelog update and also checking for the collective (should be `opencollective`). But down the line when we add the publishing updates as a collective feature (say `Open Collective Engineering` or any other collective) we will have to come back here and add more conditions for the caching to work. But I might be wrong here, in which case please let me know and I'll change it to this approach. 🤔 🐨

Reply:
`latestChangelogPublishDate` is indeed a good way to optimize this query that is going to be called very often. Calling `updates` with a `limit: 1` would be more expensive, and more difficult in the Frontend to update Apollo's cache when the changelog entry is viewed by the user. Since the feature is going to stay, I don't think it's a bad thing to have specific queries for it.

As an alternative design, we can eventually move this field to `UserType`/`Individual`, as something like `hasSeenLatestChangelogEntry`. That way we can keep a caching strategy, while reducing the number of fields since we wouldn't need to have a `User.changelogViewDate` anymore on the GraphQL API.

```es6
const UserType = {
  ...,
  fields : {
    ...,
    hasSeenLatestChangelogEntry: {
      ...,
      resolve(user, _, req) {
        const latestChangelogPublishDate = getValueFromCache();
        return user.changelogViewDate >= latestChangelogPublishDate;
      }
    }
  }
}
```

Reply:
@znarf : So as per Benjamin's comment I will leave this as it is for now if you are okay with it. But let me know if you disagree and we can work towards a different approach. 😄 🐨

Reply:
Let's wait for @znarf's feedback before closing this one

Reply:
@Betree @SudharakaP I generally prefer my proposed approach but you can move forward with the current one if you're unconvinced. Caching not that a big deal for me and could also be applied to the collection if it was necessary.

The other approach from Ben is interesting and could be a compromise.

Generally, as all options are not introducing much debt, we can later change our mind and iterate.

Reply:
@znarf @Betree : I tried moving the field `latestChangelogPublishDate` but it seems that this is not a straightforward change on the front-end side. The problem here is that we use GraphQl v1 for getting the `LoggedInUser`. And we need this to pass the `slug` (or some identifier such as `id`) for the `Individual` query. This creates some problems since we will have to first wait until the `LoggedInUser` is fully loaded and then try to get the `Individual`. I couldn't figure out a nice way of doing this. Ideally, I think we should get this user object through one single API in a single pass but that will involve migrating the User query in the front-end from GraphQl V1 to V2. Let me know what you think. 🤔 🐨 

I've opened two draft PR's to demonstrate what I mean; https://github.com/opencollective/opencollective-api/pull/6167 and https://github.com/opencollective/opencollective-frontend/pull/6501. 🤔

Note that, the problem here is, https://github.com/opencollective/opencollective-frontend/blob/eb17345c134bc0ab87af47fe9d38c2b76ec55788/components/TopBar.js#L344-L351 which depends on the LoggedInUser.

Reply:
I think the simplest is to have this `hasSeenLatestChangelogEntry` field on both `Individual` and `UserType`. Retrieving the info from `LoggedInUser` will be much simpler. But in any case, you can fetch the logged-in individual from GQLV2 using `loggedInAccount`

Reply:
@Betree : Thanks. That was helpful. 🙏🏽  

@znarf @Betree : I've handled this in the other two PRs, https://github.com/opencollective/opencollective-api/pull/6167 and https://github.com/opencollective/opencollective-frontend/pull/6501 so that this don't get too much polluted and probably easier to review. Thanks and let me know if you find any other issues. 🐨 😄